src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+  }
+
+  return (
+    <form action={requestUsername}>
+      <input type="text" name="username" />
+      <button type="submit">Request</button>
+    </form>
+  );
+}
+```
+
+The `<form>` is rendered in a Server Component but the form submission will take place on the client. Before [hydration](/reference/react-dom/client/hydrateRoot#hydrating-server-rendered-html) or if JavaScript is disabled, this form will still allow users to call the server action.
+
+#### Handling return values {/*handling-return-values*/}
+
+You'll notice that in a username request form, there might be the chance that a username is not available. `requestUsername` should tell us if it fails or not. How does the form receive the return value of the server action?
+
+Here is where you can use [`useFormState`](/reference/react-dom/hooks/useFormState), a hook designed for receiving a server action return value that supports progressive enhancement.
+
+```js
+// App.js
+import {useFormState} from 'react';
+
+export default App() {
+  function requestUsername(formData) {
+    'use server';
+    const username = formData.get('username');
+    if (canRequest(username)) {
+      // ...
+      return 'successful';
+    }
+    return 'failed';
+  }
+
+  const [returnValue, action] = useFormState(requestUsername, 'n/a');
+
+  return (
+    <>
+      <form action={requestUsername}>
+        <input type="text" name="username" />
+        <button type="submit">Request</button>
+      </form>
+      <p>Last submission request returned: {returnValue}</p>
+    </>
+  );
+}
+```
+
+### Calling a server action outside of `<form>` {/*calling-a-server-action-outside-of-form*/}
+
+Server actions are exposed server endpoints to client code and can in theory be called anywhere in client code.
+
+```js
+import incrementLike from './actions';
+import {useState, useTransition} from 'react';
+
+function LikeButton() {
+  const [pending, startTransition] = useTransition();
+  const [likeCount, setLikeCount] = useState(0);
+
+  const onClick = () => startTransition(async () => {
+    const currentCount = await incrementLike();
+    setLikeCount(currentCount);
+  });
+
+  return (
+    <>
+      <p>Total Likes: {likeCount}</p>
+      <button onClick={onClick} disabled={pending}>Like</button>;
+    </>
+  );
+}
+```
+
+If you call a server action outside of a [form](/reference/react-dom/components/form), which has first-class support for server actions, you can only do so in a Client Component, which lacks progressive enhancement. In addition, you'll need to manually call the server action in a transition `<form action>` takes care of this for you.
 
-This API can be used in any framework that supports React Server Components. You may find additional documentation from them.
-* [Next.js documentation](https://nextjs.org/docs/getting-started/react-essentials)
-* More coming soon
-</Wip>
\ No newline at end of file
+To read a server action return value, you'll need to `await` the promise returned.
```

Comment:
I am pretty sure this is right and reasonable, but I'm curious how to explain why you should be using await here and not using the `use` function. Is it because we are inside an async function already so it is fine, and also we don't want to make this component suspend while waiting for the server response, you handle that with the pending flag instead of suspense?

Reply:
Hmm that's a good callout. Yea I don't think `use` should be used in this callback..but I'm going off intuition. @sophiebits is that right? Would we ever want to call a server action in render? I guess they're always called in transitions..

Edit: Trying out `use`, I get the following error

```
Error: Hooks are not supported inside an async component. This error is often caused by accidentally adding `'use client'` to a module that was originally written for the server.
```

Reply:
use() is only to unwrap a resource during the render pass (i.e. in a component body or in a setstate updater). Not in nested functions like the imperative one here.

You shouldn't call a server action in render; refactor your code so you don't have to (plus, SSR won't let you).